nixos/modules/services/continuous-integration/buildkite-agents.nix

```diff
@@ -60,6 +60,12 @@ let
         '';
       };
 
+      user = lib.mkOption {
+        type = lib.types.passwdEntry lib.types.str;
+        default = "buildkite-agent-${name}";
```

Comment:
Where does the `name` binding come from?

Reply:
It gets passed as an argument to the submodule by the module machinery. It's the name of the attr.

Reply:
Okay, it wasn't obvious to me but I understand now, thanks!